### PR TITLE
fix(core): NativeLPSZ does not work correctly with UTF8 names

### LINKED ISSUE #1614: Web Console does not display metadata for tables with Unicode name
### Describe the bug

As [stated](https://questdb.io/docs/reference/sql/create-table/#tablename) in the documentation, table names can contain both ASCII and Unicode characters, while I've encountered an issue when creating a table with Unicode characters in the name.

### To reproduce

1. Run the following DDL in Web Console:
```SQL
CREATE TABLE 'привет от штиблет' (f0 STRING, f1 STRING, f2 STRING);
```

2. Then hit Refresh button in Tables section. As the result, the table is not shown and the following log message gets written to stdout on server side:
```
2021-11-25T09:56:50.799772Z I i.q.g.e.f.c.TableMetadataCursorFactory cannot query table metadata [table=привет от штиблет, error=File not found: /tmp/questdb/db/￐﾿￑ﾀ￐ﾸ￐ﾲ￐ﾵ￑ﾂ ￐ﾾ￑ﾂ ￑ﾈ￑ﾂ￐ﾸ￐ﾱ￐ﾻ￐ﾵ￑ﾂ/_meta, errno=0]
```

### Expected Behavior

The table is shown in the Web Console UI.

### Environment

```markdown
- **QuestDB version**: latest 6.1.3-SNAPSHOT
- **OS**: Ubuntu 20.04
- **Browser**:
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bluestreak01**
Refactoring use of this class out. Closes #1614

- removed NativeLPSZ
- refactored file system scanning code globally to use simpler UTF8 compatible algorithms
- new `notDots()` implementation that does not require `CharSequence` instance
- fixed issue with `insert into`, it was impossible to `insert into 'table name'...`. SqlParser was insisting on literal
- fixed O3 issue with not supporting non-ascii table and column names
- added tests for O3 and the original issue
- changed logging to print correct utf8 SQL and table names

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 41 / 42 (97.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|27|27|100.00%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 41 / 42 (97.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|27|27|100.00%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 147 / 158 (93.04%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Path.java|21|26|80.77%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|31|36|86.11%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/tasks/O3OpenColumnTask.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/QueryCache.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|46|46|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/tasks/O3PartitionTask.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 147 / 158 (93.04%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Path.java|21|26|80.77%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|31|36|86.11%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/tasks/O3OpenColumnTask.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/QueryCache.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|46|46|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/tasks/O3PartitionTask.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 145 / 160 (90.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|33|39|84.62%|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|34|37|91.89%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|25|26|96.15%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/QueryCache.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 145 / 160 (90.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|33|39|84.62%|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|34|37|91.89%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|25|26|96.15%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/QueryCache.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 1]
```diff
@@ -1064,7 +1099,9 @@ private ExecutionModel parseInsert(GenericLexer lexer) throws SqlException {
             throw SqlException.$(lexer.lastTokenPosition(), "'into' expected");
         }
 
-        model.setTableName(expectLiteral(lexer));
+        tok = tok(lexer, "table name");
+
+        model.setTableName(nextLiteral(GenericLexer.assertNoDotsAndSlashes(GenericLexer.unquote(tok), lexer.lastTokenPosition()), lexer.lastTokenPosition()));
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:1104]: We have `TableUtils.isValidColumnName` and it's more strict than just dots and slashes. I believe same level of validation should be for table names as it is for column names

File: core/src/test/java/io/questdb/cutlass/http/IODispatcherTest.java [hunk 2]
```diff
@@ -27,6 +27,9 @@
 import static io.questdb.test.tools.TestUtils.assertMemoryLeak;
 
 import java.io.InputStream;
+import java.time.LocalDateTime;
```
Comment (ideoma) [inline core/src/test/java/io/questdb/cutlass/http/IODispatcherTest.java:30]: Looks like import changes with no reason

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java [hunk 3]
```diff
@@ -192,20 +192,20 @@ public void testBindVariableInIndexLookup() throws Exception {
...
         assertMemoryLeak(() -> {
             compiler.compile("CREATE TABLE 'alcatel_traffic_tmp' (" +
-                    "deviceName SYMBOL capacity 1000 index, " +
-                    "time TIMESTAMP, " +
-                    "slot SYMBOL, " +
-                    "port SYMBOL, " +
-                    "downStream DOUBLE, " +
-                    "upStream DOUBLE" +
-                    ") timestamp(time) partition by DAY",
+                            "deviceName SYMBOL capacity 1000 index, " +
```
Comment (ideoma) [inline core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:195]: Too many changes in the file, hard to see if new tests are added.

File: core/src/test/java/io/questdb/test/tools/TestUtils.java [hunk 4]
```diff
@@ -38,6 +38,7 @@
 import io.questdb.network.NetworkFacadeImpl;
 import io.questdb.std.*;
 import io.questdb.std.datetime.microtime.Timestamps;
+import io.questdb.std.str.DirectUnboundedByteSink;
```
Comment (ideoma) [inline core/src/test/java/io/questdb/test/tools/TestUtils.java:41]: no changes but import

**Review by ideoma — APPROVED** [review]
Approved with minor comments

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java (hunk 3 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:195]: sorry, this is auto-formatting issue. One test was added `testUtf8TableName()`

File: core/src/test/java/io/questdb/cutlass/http/IODispatcherTest.java (hunk 2 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/cutlass/http/IODispatcherTest.java:30]: intellij formatting :(

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/SqlParser.java:1104]: we need to tidy this up separately, the above is table name - column name requirements are different

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 145 / 160 (90.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|33|39|84.62%|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|34|37|91.89%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|25|26|96.15%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/QueryCache.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|

File: core/src/test/java/io/questdb/test/tools/TestUtils.java (hunk 4 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/test/tools/TestUtils.java]: fixed

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java (hunk 3 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:195]: fixed in https://github.com/questdb/questdb/pull/1623/commits/0186f92481bb59cae4fdd01e7ecfff707800cb49

File: core/src/test/java/io/questdb/cutlass/http/IODispatcherTest.java (hunk 2 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/cutlass/http/IODispatcherTest.java]: fixed in https://github.com/questdb/questdb/pull/1623/commits/0186f92481bb59cae4fdd01e7ecfff707800cb49

File: core/src/main/java/io/questdb/std/Files.java [hunk 5]
```diff
@@ -201,6 +202,41 @@ public static void munmap(long address, long len, int memoryTag) {
         }
     }
 
+    public static boolean notDots(CharSequence value) {
+        final int len = value.length();
+        if (len > 2) {
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/std/Files.java:207]: this method is only used with `findFile`, cannot be empty

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 145 / 160 (90.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cairo/mig/EngineMigration.java|33|39|84.62%|
|:large_blue_circle:|io/questdb/cairo/O3PurgeDiscoveryJob.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|34|37|91.89%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|25|26|96.15%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/QueryCache.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**